db/migrate/20231213092233_add_devise_to_users.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class AddDeviseToUsers < ActiveRecord::Migration[7.1]
+  def self.up
+    change_table :users do |t|
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
+
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
+      # t.integer  :sign_in_count, default: 0, null: false
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.string   :current_sign_in_ip
+      # t.string   :last_sign_in_ip
+
+      ## Confirmable
+      t.string   :confirmation_token
+      t.datetime :confirmed_at
+      t.datetime :confirmation_sent_at
+      t.string   :unconfirmed_email # Only if using reconfirmable
+
```

Comment:
- [ ] When I try to run your migration files, I get the following error below in the log produced in the terminal:

![image](https://github.com/Megagig/Blog-App/assets/77106962/e9ccf5ae-6b77-41db-8dff-41ca2d02be62)

---
Since you have uncommented the highlighted piece of code, you don't need this migration file:

`db/migrate/20231213135623_add_confirmable_to_devise.rb`

It would seem like a duplication and thus produce the error shown in the screenshot above. I would advise you to delete this migration file `db/migrate/20231213135623_add_confirmable_to_devise.rb`. 👍 😎